lib/ansible/modules/cloud/amazon/kinesis_firehose.py

```diff
@@ -0,0 +1,626 @@
+          enabled: true
+          processors:
+            -
+              type: Lambda
+              parameters:
+                -
+                  parameter_name: LambdaArn
+                  parameter_value: "arn:aws:lambda:eu-west-1:XXXXXXXXXXXX:function:RecordDelimiter"
+      wait: yes
+      wait_timeout: 600
+      state: present
+    register: testout
+
+  # Basic delete example:
+  - name: Delete Kinesis Firehose Stream
+    kinesis_firehose:
+      delivery_stream_name: clarky-test-stream
+      delivery_stream_type: KinesisStreamAsSource
+      kinesis_stream_source_configuration:
+      s3_destination_configuration:
+      wait: yes
+      wait_timeout: 300
+      state: absent
+    register: testout
+'''
+
+RETURN = '''
+delivery_stream_name:
+  description: The name of the Kinesis Firehose Delivery Stream.
+  returned: when state == present.
+  type: string
+  sample: "test-stream"
+delivery_stream_arn:
+  description: The amazon resource identifier
+  returned: when state == present.
+  type: string
+  sample: "arn:aws:kinesis:east-side:123456789:stream/test-stream"
+delivery_stream_status:
+  description: The current state of the Kinesis Firehose Delivery Stream.
+  returned: when state == present.
+  type: string
+  sample: "ACTIVE"
+'''
+
+try:
+    import botocore
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import logging
+import re
+import datetime
+import time
+from ansible.module_utils._text import to_native
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict, ec2_argument_spec, get_aws_connection_info
+
+
+def convert_to_snake_case(data):
+    """Convert all uppercase keys in dict with lowercase_
+    Args:
+        data (dict): Dictionary with keys that have upper cases in them
+            Example.. FooBar == foo_bar
+            if a val is of type datetime.datetime, it will be converted to
+            the ISO 8601
+
+    Basic Usage:
+        >>> test = {'FooBar': []}
+        >>> test = convert_to_snake_case(test)
+        {
+            'foo_bar': []
+        }
+
+    Returns:
+        Dictionary
+    """
+    results = dict()
+    if isinstance(data, dict):
+        for key, val in data.items():
+            key = re.sub(r'(([A-Z]{1,3}){1})', r'_\1', key).lower()
+            if key[0] == '_':
+                key = key[1:]
+            if isinstance(val, datetime.datetime):
+                results[key] = val.isoformat()
+            elif isinstance(val, dict):
+                results[key] = convert_to_snake_case(val)
+            elif isinstance(val, list):
+                converted = list()
+                for item in val:
+                    converted.append(convert_to_snake_case(item))
+                results[key] = converted
+            else:
+                results[key] = val
+    return results
+
+
+def convert_to_title_case(data):
```

Comment:
`snake_dict_to_camel_dict` already exists in module_utils.ec2, it has recently been updated to allow a second parameter `capitalize_first=True` (because someone made the default dromedaryCase early on)